### PR TITLE
feat(platform): support dynamic registry

### PR DESCRIPTION
**Pull request by leoryu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by huxiaoliang — APPROVED** [review]

File: api/platform/types.go [hunk 1]
```diff
@@ -123,6 +123,8 @@ const (
 	AnywhereApplicationAnno = "tkestack.io/anywhere-application"
 	// AnywhereValidateAnno is exist, the cluster will always return validate result
 	AnywhereValidateAnno = "tkestack.io/anywhere-validate"
+	// AnywhereImagePrefixAnno is exist, the cluster will use it as k8s images prefix
+	AnywhereImagePrefixAnno = "tkestack.io/anywhere-image-prefix"
```
Comment (huxiaoliang) [inline api/platform/types.go:127]: location-based-image-prefix

File: pkg/platform/provider/baremetal/cluster/update.go [hunk 2]
```diff
@@ -84,7 +85,7 @@ func (p *Provider) EnsureRenewCerts(ctx context.Context, c *v1.Cluster) error {
 }
 
 func (p *Provider) EnsureAPIServerCert(ctx context.Context, c *v1.Cluster) error {
-	kubeadmConfig := p.getKubeadmInitConfig(c)
+	kubeadmConfig := p.GetKubeadmInitConfig(c, containerregistry.GetPrefix())
```
Comment (wl-chen) [inline pkg/platform/provider/baremetal/cluster/update.go:88]: 这里直接用的原来的repo吗？

File: pkg/platform/provider/baremetal/cluster/kubeadm.go [hunk 3]
```diff
@@ -184,7 +184,7 @@ func (p *Provider) getClusterConfiguration(c *v1.Cluster) *kubeadmv1beta2.Cluste
 		DNS: kubeadmv1beta2.DNS{
 			Type: kubeadmv1beta2.CoreDNS,
 		},
-		ImageRepository: containerregistry.GetPrefix(),
+		ImageRepository: repo,
```
Comment (wl-chen) [inline pkg/platform/provider/baremetal/cluster/kubeadm.go:187]: 这里是不是能进行收敛，直接用这个 p.getImagePrefix(c)

File: pkg/platform/provider/baremetal/cluster/kubeadm.go [hunk 4]
```diff
@@ -309,9 +309,9 @@ func (p *Provider) getSchedulerExtraArgs(c *v1.Cluster) map[string]string {
 	return args
 }
 
-func (p *Provider) getKubeletExtraArgs(c *v1.Cluster) map[string]string {
+func (p *Provider) GetKubeletExtraArgs(c *v1.Cluster, repo string) map[string]string {
 	args := map[string]string{
-		"pod-infra-container-image": images.Get().Pause.FullName(),
+		"pod-infra-container-image": path.Join(repo, images.Get().Pause.BaseName()),
```
Comment (wl-chen) [inline pkg/platform/provider/baremetal/cluster/kubeadm.go:314]: 这里是不是能进行收敛，直接用这个 p.getImagePrefix(c)

**Review by wl-chen — CHANGES_REQUESTED** [review]
cc

**Review by wl-chen — APPROVED** [review]

**Pull request merged**